lang_python.adoc

```diff
@@ -13,130 +13,167 @@ To generate Python modules, Kaitai Struct compiler should be invoked with
+
+If you are using snapshot KSC, you should also use snapshot runtime.
 
 [[add-runtime-release]]
 ==== Using release version
 
-With stable version of Kaitai Struct compiler, one can use use following
+With stable version of Kaitai Struct compiler, one can use the following
 methods:
 
 * On Debian / Ubuntu, one can install it from apt repositories:
 
-** For Python 2.x:
+** For Python 3.x:
 +
 [source,shell]
-sudo apt-get install python-kaitaistruct
+sudo apt-get install -y python3-kaitaistruct
 +
-Reference: https://packages.debian.org/buster/python-kaitaistruct[package info for Debian], https://packages.ubuntu.com/eoan/python-kaitaistruct[package info for Ubuntu]
+Reference: https://packages.debian.org/testing/python3-kaitaistruct[image:https://repology.org/badge/version-for-repo/debian_testing/python:python-kaitaistruct.svg[package info for Debian]], https://packages.ubuntu.com/focal/python3-kaitaistruct[image:https://repology.org/badge/version-for-repo/ubuntu_20_04/python:python-kaitaistruct.svg[package info for Ubuntu]]
 
-** For Python 3.x:
+** For Python 2.x:
 +
 [source,shell]
-sudo apt-get install python3-kaitaistruct
+sudo apt-get install -y python-kaitaistruct
 +
-Reference: https://packages.debian.org/buster/python3-kaitaistruct[package info for Debian], https://packages.ubuntu.com/eoan/python3-kaitaistruct[package info for Ubuntu]
+Reference: https://packages.debian.org/stable/python-kaitaistruct[image:https://repology.org/badge/version-for-repo/debian_stable/python:python-kaitaistruct.svg[package info for Debian]], https://packages.ubuntu.com/focal/python-kaitaistruct[image:https://repology.org/badge/version-for-repo/ubuntu_20_04/python:python-kaitaistruct.svg[package info for Ubuntu]]
 
-* If your project is using *requirements.txt* to manage your
-  dependencies, add the following line to it:
+
+* Add the following dependency specifier into the package configuration
+of the packaging tool you use:
 +
 ----
 kaitaistruct
 ----
 +
-and then run `pip install -r requirements.txt` to update all your
-dependencies.
+and then run `python3 -m pip install --upgrade -e ./` in the dir with your project
+in order to install your package and all its dependencies in
+an editable way. If you don't want to package your software,
+but only install its dependencies, add `kaitaistruct` into `requirements.txt`.
 
-* Otherwise, you can just install the package manually using *pip*
+* Otherwise, you can just install the package manually using `pip`
   (Python package manager):
 +
 [source,shell]
-pip install kaitaistruct
+python3 -m pip install --upgrade kaitaistruct
 
 [[add-runtime-snapshot]]
 ==== Using latest (snapshot) version
 
-If you're using latest (unreleased AKA "unstable" AKA "snapshot") build
-of Kaitai Struct compiler, that will require latest runtime libraries as
-well:
+If you're using latest (unreleased AKA "bleeding edge" AKA "unstable"
+AKA "snapshot" AKA "nightly") build of Kaitai Struct compiler, that will
```

Comment:
I don't really like using the _AKA_ abbreviation that many times as the delimiter of the alternatives, what about using commas and _or_ before the last one? Another thing that I'm not glad to see is the a.k.a. abbreviation written without dots with all letters uppercase. I remember that when I first encountered it in the KS docs, I had to search it by Google what it's supposed to mean (if it's some special technical thing), because I've never seen it written other than like `a.k.a.` before. The uppercase acronyms are generally more often used for artificial names like JSON, XML, JPEG, ..., not for common "language constructs" like etc. (_et cetera_), e.g. (_exempli gratia_), ca. (_circa_), and so on.

The best option is of course not to use the abbreviations at all to avoid confusion.

```suggestion
If you're using latest (unreleased, also known as "bleeding edge", "unstable",
"snapshot" or "nightly") build of Kaitai Struct compiler, that will
```

Reply:
The traditional spelling is without dots:

[![https://writingexplained.org/aka-or-aka](https://writingexplained.org/wp-content/uploads/definition-of-aka-definition-.png)](https://writingexplained.org/aka-or-aka)

Reply:
OK, thanks for the stats. But I don't think that the uppercase variant is too frequent, the letter case is my main concern. And it doesn't change the fact that it's generally better to use the full variant whenever possible.

Reply:
`aka` is not really used in its expanded variant.

Reply:
> `aka` is a kind of a meme and is not really used in its expanded variant.

BTW, I don't really think so, see this random page on Wikipedia for example: https://en.wikipedia.org/wiki/Splitting_maul

> A **splitting maul** also known as a _block buster_, _block splitter_, _chop and maul_, _sledge axe_, _go-devil_ or _hamaxe_ is a heavy, long-handled axe used for splitting a piece of wood along its grain.

Reply:
Usage of memes is a higly informal style, it is not the style accepted on wikipedia. It is more accepted in dramatica and other places rather than on wikipedia.

Reply:
> Usage of memes is a higly informal style, it is not the style accepted on wikipedia. It is more accepted in dramatica and other places rather than on wikipedia.

If this style is not accepted by Wikipedia, why it should be accepted by the official Kaitai Struct documentation? Are the KS docs somewhat more inferior compared to Wikipedia? 

I think we should better try to make the docs look more professionally. _Open source_ doesn't imply _less significant_ or _more informal_. Our official documentation should not look like a chat.

On top of that, Kaitai Struct is used by users from around the world, and many of them don't have English as a native language. Using informal language makes it harder for these people to understand, because they're usually not familiar with the colloquial English used by native speakers for everyday communication.

Reply:
>I think we should better try to make the docs look more professionally.

Are you trying to raise funding from some sponsor, who demands to remove it? If not, why should we?

>Are the KS docs somewhat more inferior compared to Wikipedia?

No, it is Wikipedia that inferior to less formal encyclopædias in some sense. Some facts cannot go to wikipedia because of WP rules about what WP is for and what is it not for. You probably shouldn't apply WP rules to the rest of the world.

>Our official documentation should not look like a chat.

It doesn't.

>On top of that, Kaitai Struct is used by users from around the world, and many of them don't have English as a native language.

The first time I saw this abbreviation, it was in a magazine (printed on paper) written in Russian around 2004. Though I have never met the abbreviation before, I quickly derived from the context what it means.

Reply:
> Are you trying to raise funding from some sponsor, who demands to remove it? If not, why should we?

You don't need a multi-million dollar budget to change a single sentence to comply with the formal language and style.

Why should we do it? To make the docs more readable, understandable and clear, to make the docs look reputable, and also not to discourage potential users by the bad condition or language level of the docs. 

> No, it is Wikipedia that inferior to less formal encyclopædias in some sense. Some facts cannot go to wikipedia because (...)

I'm sorry, did I say a word about _facts_? I've only used Wikipedia as an example site from which we can still learn a few things regarding to language and style of the text in my opinion. Wikipedia content may be inaccurate or incorrect, but I don't see any reason why we should discuss it here. That's a whole different matter.

If you consider the language and style level on Wikipedia still too low in some sense, please find a more reputable site which uses literally _AKA_ or _aka_ and share a link to it.

> > Our official documentation should not look like a chat.
> 
> It doesn't.

Using _aka_ four times in a row seems more appropriate rather for an informal chat than for any official documentation. It's like when I write _"I need to buy apples and bread and ham and milk and meat."_. You don't need to be a linguist to notice that something is wrong.

> The first time I saw this abbreviation, it was in a magazine (printed on paper) written in Russian around 2004. Though I have never met the abbreviation before, I quickly derived from the context what it means.

You wouldn't have to derive anything if the magazine used the expanded "also known as" in Russian instead. Besides, even if the particular _aka_ word was understandable by 100% of our audience, it lowers the overall language style and feeling of reputability of the docs.

Reply:
>it lowers the overall language style

It is actually a good thing.

>feeling of reputability of the docs

We are neither a parliament, nor an army headquaters, nor a court. We are just a group of people working on a cool tool. And this style brings more reputability and trust then trying to look more serious than we are.

Reply:
> > it lowers the overall language style
>
> It is actually a good thing.

> And this style brings more reputability and trust then trying to look more serious than we are.

I really doubt that, but I can see that I'm not going to move the mountain. 
I hope you're right, as I'm not willing to waste any more hours in this discussion. Let's focus on some stuff that matters.